R/expect_comparison_linter.R

```diff
@@ -0,0 +1,44 @@
+#' Require usage of expect_gt(x, y) over expect_true(x > y) (and similar)
+#'
+#' [testthat::expect_gt()], [testthat::expect_gte()], [testthat::expect_lt()],
+#'   [testthat::expect_lte()], and [testthat::expect_equal()] exist specifically
+#'   for testing comparisons between two objects. [testthat::expect_true()] can
+#'   also be used for such tests, but it is better to use the tailored function
+#'   instead.
+#'
+#' @evalRd rd_tags("expect_comparison_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+expect_comparison_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$parsed_content) == 0L) {
+      return(list())
+    }
+
+    xml <- source_file$xml_parsed_content
+
+    # != doesn't have a clean replacement
+    comparator_nodes <-
+      setdiff(as.list(infix_metadata$xml_tag[infix_metadata$comparator]), "NE")  # nolint: object_usage_linter.
+    xpath <- glue::glue("//expr[
+      expr[SYMBOL_FUNCTION_CALL[text() = 'expect_true']]
+      and expr[2][ {do.call(xp_or, comparator_nodes)} ]
```

Comment:
better extend `xp_or()` to be vectorized?

Reply:
I'll write that as a separate PR. I had started to do that for another PR but it did look messy. Probably worth it though.

Reply:
#961